### PR TITLE
Blog - Building a Real-time Stock Price Dashboard Using Python, QuestDB, and Plotlye

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
Hey @jalletto could you please help on this.

**Comment by jalletto** [comment]
Hi @msali123. Looks like this is failing because of white space on line 133. I think it's just a blank space that needs to be deleted. 

`         ./blog+lint *failed* | ./blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:133:1 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]`

**Comment by msali123** [comment]
@jalletto finally done 🤩

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+</div>
+<div class="wide">
+![Accessing QuestDB's web console]({{site.images}}{{page.slug}}/IyA9iv1.png)
+</div>
+
+### Creating the Database Table
+
+Now that you have QuestDB running, you can create the database table by connecting to QuestDB's interactive console and running the below SQL statement.
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:99]:
```suggestion
~~~{.sql }
```

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+
+<div class="wide">
+![Creating QuestDB Table]({{site.images}}{{page.slug}}/CkwHRFz.png)
+</div>
+
+## Creating Workers Using Celery
+
+In this section, you will define Python dependencies that the project requires, define the worker settings, and create the periodic tasks that will fetch data from Finnhub.
+
+![Creating Workers Using Celery]({{site.images}}{{page.slug}}/lqv7XJ9.png)
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:132]:
```suggestion
![Creating Workers Using Celery]({{site.images}}{{page.slug}}/lqv7XJ9.png)\
```

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+# [...]
+ 
+    # Database settings
+    database_url: str = "postgresql://admin:quest@127.0.0.1:8812/qdb"
+    database_pool_size: int = 3
+
+# [...]
+~~~
+
+![Image]({{site.images}}{{page.slug}}/p5JXM6g.png)
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:229]:
```suggestion
![Image]({{site.images}}{{page.slug}}/p5JXM6g.png)\
```

Comment (jalletto) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:229]: Let's add the citation for the image if we have it.

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+        # used by the OS or other software.
+        env_prefix = "SMD_"  # SMD stands for Stock Market Dashboard
+ 
+ 
+settings = Settings()
+~~~
+
+In the above `settings.py` file, `celery_broker` and `database_url` settings have been defined with default values. You will have to change their values with the correct settings and run the worker in a Docker container to get started with the settings. To keep the environment separated, you will create a `.env` file in the project root directory with the below content. The reason for using a `.env` file is to store sensitive credentials such as API keys, the stock symbols, and the frequency at which stock prices are fetched from the API.
+
+~~~{.env caption=""}
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:277]:
```suggestion
~~~{ caption=".env"}
```

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+
+This tutorial focuses less on the front-end of the application—meaning you won't be required to write stylesheets for the application—but just copy and paste existing code. You will create an `assets` directory in the root project directory and download the `styles.css` file using the `curl` command below. This is possible as Dash provides you with boilerplate code.
+
+~~~{.bash caption=">_"}
+$ curl -o assets/style.css https://github.com/verah-tech/stock-market/blob/main/assets/style.css
+~~~
+
+### Setting Up the Application
+
+![Setting Up the Application]({{site.images}}{{page.slug}}/3OFABqK.png)
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:420]:
```suggestion
![Setting Up the Application]({{site.images}}{{page.slug}}/3OFABqK.png)\
```

**Review by jalletto — CHANGES_REQUESTED** [review]

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md (hunk 3 above)
Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:229]: @jalletto these images were provided by author so should I remove them let them go without credit?

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md (hunk 5 above)
Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:420]: provided by author
@jalletto

Comment (jalletto) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: @msali123 Can you email the author and ask if they have a source?

Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: Okay I'll do

Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: image

Should we remove these images? @jalletto

Comment (jalletto) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: No we only need citations for graphics, not screen shots. I might not have been clear about that. Sorry about the confusion.

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+~~~
+
+### Installing QuestDB and Redis
+
+Next you need to install the project's required services. In this project, you need Docker and [Docker Compose](/blog/youre-using-docker-compose-wrong) to avoid overloading the local system. Inside the project's root directory, create a new `.yml` file named `docker-compose.yml` file for Docker Compose as shown below. The `.yml` file lets you define the configuration or [container](/blog/docker-slim) environment for multiple Docker containers.. In this project it will allow you to run QuestDB and Redis services simultaneously. Later on, you will install other services as needed.
+
+~~~{.yaml caption="docker-compose.yml"}
+# docker-compose.yml
+ 
+version: "3"
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:62]:
```suggestion
~~~{.yaml caption="docker-compose.yml"}

version: "3"
```

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md [hunk 7]
```diff
@@ -0,0 +1,725 @@
...
+
+In this section, you will set up the front end of the application using Plotly and Dash. Plotly comes with a framework known as Dash, enabling you to create a web application quickly and efficiently. Dash abstracts away the boilerplate useful in setting up a web server and creating several handles for the application.
+
+### Getting Static Assets
+
+This tutorial focuses less on the front-end of the application—meaning you won't be required to write stylesheets for the application—but just copy and paste existing code. You will create an `assets` directory in the root project directory and download the `styles.css` file using the `curl` command below. This is possible as Dash provides you with boilerplate code.
+
+~~~{.bash caption=">_"}
+$ curl -o assets/style.css https://github.com/verah-tech/stock-market/blob/main/assets/style.css
+~~~
```
Comment (jalletto) [inline blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md:416]:
```suggestion
~~~{.bash caption=">_"}

$ curl -o assets/style.css https://github.com/verah-tech/stock-market/blob/main/assets/style.css
~~~
```

File: blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md (hunk 5 above)
Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: @jalletto so I should remove the photos & make alternate designs on canva

Comment (msali123) [reply blog/_posts/2023-02-24-dashboard-with-python-questdb-plotly.md]: ??

**Comment by jalletto** [comment]
@msali123 Sorry I think there was some confusion on this one. We won't ever need sources for screen shots the author took themselves. I was talking about these images:

blog/assets/images/dashboard-with-python-questdb-plotly/3OFABqK.png
blog/assets/images/dashboard-with-python-questdb-plotly/p5JXM6g.png

Which we should site. 

Also, the title needs to updated in the header image. No 'e' at then end of Plotly. Since you are editing it, let's take this opportunity to shorten the title to: Stock Price Dashboard with QuestDB and Plotly

**Comment by msali123** [comment]
So @jalletto should I remove these 2 images?

Or make my self something relevant like this on canva?

blog/assets/images/dashboard-with-python-questdb-plotly/3OFABqK.png
blog/assets/images/dashboard-with-python-questdb-plotly/p5JXM6g.png

**Comment by balapriyac** [comment]
@jalletto @msali123 
This article is flagged for plagiarism. https://questdb.io/blog/2021/11/01/plotly-finnhub-realtime-dashboard/ is the original article published in Nov 2021. I've sent a message on Slack.

**Comment by adamgordonbell** [comment]
Unfortunately, we won't be merging this article. Pause work on this PR.

**Pull request closed without merging**